packages/DataGrid/src/DataGrid.js

```diff
@@ -0,0 +1,628 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { VariableSizeGrid as Grid } from "react-window";
+import extractChildren from "@paprika/helpers/lib/extractChildren";
+import Spinner from "@paprika/spinner";
+import Bubbly from "@paprika/icon/lib/BubblySearchNoResults";
+import Cell from "./components/Cell";
+import useGridEventHandler from "./hooks/useGridEventHandler";
+import ColumnDefinition from "./components/ColumnDefinition";
+import * as styled from "./DataGrid.styles";
+import WhenScrollBarReachedBottom, { End } from "./components/WhenScrollBarReachedBottom";
+import InfinityScroll from "./components/InfinityScroll";
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+  data: PropTypes.arrayOf(PropTypes.shape({})),
+  hasNoRecords: PropTypes.bool,
+  height: PropTypes.number,
+  isIdle: PropTypes.bool,
+  onClick: PropTypes.func,
+  onEnter: PropTypes.func,
+  onKeyDown: PropTypes.func,
+  onRowChecked: PropTypes.func,
+  onShiftSpaceBar: PropTypes.func,
+  onSpaceBar: PropTypes.func,
+  rowHeight: PropTypes.number,
+  width: PropTypes.number,
+};
+const defaultProps = {
+  data: [],
+  hasNoRecords: false,
```

Comment:
I'd prefer `hasRecords`

Reply:
I'm wondering why this prop is necessary at all, is it not equivalent to `data.length === 0`?
(otherwise I prefer `false` for default boolean values per discussion in https://github.com/acl-services/paprika/pull/399#pullrequestreview-347563964 and maybe this can be called `isEmpty` instead)

Reply:
Realizing I picked a misleading name for this prop, what actually does this props is to show a screen with bubbly saying no record found, similar to the screen `isIdle`. 

Maybe I could splitted this into different subcomponent like:

```js
<DataTable>
  <DataTable.NoRecordFound isVisible={data.length} />
  <DataTable.IsIdle isVisible={isIdle} />
</DataTable>
```